llvm/lib/ObjectYAML/ELFEmitter.cpp

```diff
@@ -1810,7 +1810,7 @@ void ELFState<ELFT>::writeSectionContent(Elf_Shdr &SHeader,
     break;
   default:
     reportError(Section.Name + ": invalid alignment for a note section: 0x" +
-                Twine::utohexstr(SHeader.sh_addralign));
+                Twine::utohexstr(Section.AddressAlign));
```

Comment:
Possibly not worth it, but I wonder if a test is in order to highlight which alignment is actually used (i.e. `Section.AddressAlign`, not `SHeader.sh_addralign`)?

Reply:
I don't think a test can check this because `Sec->AddressAlign` is assigned to `SHeader.sh_addralign` in `ELFState<ELFT>::initSectionHeaders()` just a few lines before `writeSectionContent()` is called.

Reply:
Sorry, I posted this comment in the wrong place by accident, so it might be confusing things a little. I actually was referring to the switch a bit above, not specifically the error message. In other words, showing that if `ShAddrAlign: 4` is specified but `AddressAlign: 8` is specified, the 8 is used for padding purposes. In other words, catching the problem I pointed out earlier and that got changed.

Reply:
Well, `overrideFields()` is called after `writeSectionContent()`, so even if there are different values in `ShAddrAlign` and `AddressAlign` properties, `writeSectionContent()` will still see the same value in `Section.AddressAlign` and `SHeader.sh_addralign`, taken from `AddressAlign`. Nevertheless, I've added a new test for this scenario. It even shows that you can generate a note section with an invalid alignment without triggering an error, as long as an expected alignment is specified in the `AddressAlign` property.